### PR TITLE
MCR-2891 Object History Permission

### PR DESCRIPTION
**Pull request by Hammer1279**
- add history view check to listRevisions
- add history read check to requestVersionedObject

[Link to jira](https://mycore.atlassian.net/browse/MCR-2891).

### CODE REVIEW

File: mycore-base/src/main/java/org/mycore/datamodel/common/MCRXMLMetadataManager.java [hunk 1]
```diff
@@ -164,7 +166,8 @@ public MCRContent retrieveContent(MCRObjectID mcrid, String revision) throws IOE
      * @see MCRXMLMetadataManagerAdapter#listRevisions(MCRObjectID)
      */
     public List<? extends MCRAbstractMetadataVersion<?>> listRevisions(MCRObjectID id) throws IOException {
-        return IMPLEMENTATION.listRevisions(id);
+        return MCRAccessManager.checkPermission(id, MCRAccessManager.PERMISSION_HISTORY_VIEW)
+            ? IMPLEMENTATION.listRevisions(id) : Collections.emptyList();
```
[the remark this answers is not in the document]
Comment (Hammer1279) [reply mycore-base/src/main/java/org/mycore/datamodel/common/MCRXMLMetadataManager.java]: But this is a read only method?

[the remark this answers is not in the document]
Comment (rsteph-de) [reply mycore-base/src/main/java/org/mycore/datamodel/common/MCRXMLMetadataManager.java]: Do not check Permissions in MCRXMLMetadataManager
Permission check should be implemented where the method is called (REST-API, URI-Resolver, MCRObjectServlet, ???)

File: mycore-base/src/main/java/org/mycore/frontend/servlets/MCRObjectServlet.java [hunk 2]
```diff
@@ -100,6 +101,14 @@ private MCRContent requestLocalObject(MCRObjectID mcrid, final HttpServletRespon
 
     private MCRContent requestVersionedObject(final MCRObjectID mcrid, final HttpServletResponse resp, final String rev)
         throws IOException {
+        if (!MCRAccessManager.checkPermission(mcrid, PERMISSION_HISTORY_READ)) {
+            final MCRSession currentSession = MCRSessionMgr.getCurrentSession();
+            resp.sendError(
+                HttpServletResponse.SC_UNAUTHORIZED,
+                getErrorI18N(I18N_ERROR_PREFIX, "accessDenied", mcrid.toString() + " " + rev,
```
Comment (rsteph-de) [inline mycore-base/src/main/java/org/mycore/frontend/servlets/MCRObjectServlet.java]: use a better name for "accessDenied", something like "accessToVersionDenied"
and provide a proper error message

**Review by rsteph-de — CHANGES_REQUESTED** [review]
Sebastian and I looked into the code, I will add more implementation notes to the JIRA-ticket

**Comment by Hammer1279** [comment]
~~I think I'll have to recreate the branch... this looks very wrong...~~
Resolved, but leaving this here to explain what happened.

**Comment by Hammer1279** [comment]
Note for the missing i18n for anything except for german and english:
While I am able to edit the i18n, it would ruin the entire formatting of all those files, so I had to edit them by hand instead of with tools.
That raises the question if it is important to have them formatted to be easily human readable if we mostly use tools anyways to edit them or if it is okay to have them all bunched together.

**Review by sebhofmann — APPROVED** [review]

**Review by rsteph-de — APPROVED** [review]
looks good now after final changes  (acl:checkPermission in versions.xml)

**Pull request merged**